src/helpers/openingHoursHelper.ts

```diff
@@ -0,0 +1,299 @@
+      // in that case the current interval needs to be checked in the next step of reducing the array
+      mergedIntervals.push(previousInterval, currentInterval);
+
+      return mergedIntervals;
+    },
+    [sortedIntervals[0]]
+  );
+};
+
+/**
+ * Finds the next interval after, or surrounding the time, that is open, but not closed.
+ * It assumes that both the openIntervals and the closedIntervals are consisting of sorted and disjoint intervals.
+ */
+const findNextOpenTime = (
+  openIntervals: OpeningInterval[],
+  closedIntervals: OpeningInterval[],
+  time: Date
+): Date | undefined => {
+  const opens = openIntervals.filter((interval) => !interval.timeTo || interval.timeTo > time);
+
+  if (!opens.length) {
+    return;
+  }
+
+  const closed = closedIntervals.find((interval) =>
+    dateIsWithinInterval(time, interval.timeFrom, interval.timeTo)
+  );
+
+  if (closed) {
+    if (!closed.timeTo) {
+      return;
+    }
+
+    return findNextOpenTime(opens, closedIntervals, closed.timeTo);
+  }
+
+  return opens[0].timeFrom && opens[0].timeFrom > time ? opens[0].timeFrom : time;
+};
+
+/**
+ * Finds the next interval after, or surrounding the time, that is not open or closed.
+ * It assumes that both the openIntervals and the closedIntervals are consisting of sorted and disjoint intervals.
+ */
+const findNextClosedTime = (
+  openIntervals: OpeningInterval[],
+  closedIntervals: OpeningInterval[],
+  time: Date
+) => {
+  const openInterval = openIntervals.find((interval) =>
+    dateIsWithinInterval(time, interval.timeFrom, interval.timeTo)
+  );
+  const closedInterval = closedIntervals.find((interval) =>
+    dateIsWithinInterval(time, interval.timeFrom, interval.timeTo)
+  );
+
+  // if it is open and not closed, find the next closed time and compare if that is sooner than the open time expiring
+  if (openInterval && !closedInterval) {
+    const upcomingClosedTime = closedIntervals.find(
+      (interval) => !!interval.timeFrom && interval.timeFrom > time
+    )?.timeFrom;
+
+    if (upcomingClosedTime && openInterval.timeTo) {
+      return upcomingClosedTime < openInterval.timeTo ? upcomingClosedTime : openInterval.timeTo;
+    }
+
+    return openInterval.timeTo || upcomingClosedTime;
+  }
+
+  // if it is either closed or not open, the time given is already a closed time.
+  return time;
+};
+
+/**
+ * WARNING: testDate is only for tests, and will yield unexpected result if the day is not the present day.
+ *
+ * Checks for a given array of opening times, if it is currently open, and if it opens or closes soon.
+ * The timeDiff contains the remaining minutes until it opens or closes,
+ * timeDiff may be undefined, if there are opening times where either timeFrom or timeTo is missing.
+ */
+// eslint-disable-next-line complexity
+export const isOpen = (
+  openingHours: OpeningHour[],
+  testDate?: Date
```

Comment:
i have problems with this "test-only" parameter we need to take care of and describe in comments. what about making the `now` our parameter here with `new Date()` as default value? when calling this method in tests, we could pass different values. this way we have no specific case we need to warn about and think of, right? we can "just use it" the way we need the method, right?

Reply:
i changed `testDate` to now with fallback of `new Date()`,  and changed the warning. ( the date passed still needs to be "today" because the behaviour will not be as expected otherwise.